docs/src/main/asciidoc/se/injection.adoc

```diff
@@ -0,0 +1,850 @@
+Default name is `ApplicationBinding` (customizable)
+
+Methods that accept the bindings are on
+link:{service-registry-base-url}/io/helidon/service/registry/ServiceRegistryManager.html[`ServiceRegistryManager`]:
+
+- `start(Binding)` - starts the service registry with the generated binding,
+initializing all singleton and per-lookup services annotated with a
+link:{service-registry-base-url}/io/helidon/service/registry/Service.RunLevel.html[`@Service.RunLevel`]
+annotation (i.e. `start(ApplicationBinding.create())`)
+- `start(Binding, ServiceRegistryConfig)` - same as above, allows for customization of configuration,
+if used, remember to set discovery to `false` to prevent automated discovery from the classpath
+
+All options to start a Helidon application that uses service registry:
+
+- A custom Main method using `ServiceRegistryManager.start(...)` methods, or `ServiceRegistryManager.create(...)` methods
+- A generated `ApplicationMain` - optional feature of the Maven plugin, requires property `generateMain` to be set to `true`.
+It uses link:{service-registry-base-url}/io/helidon/service/registry/Service.RunLevel.html[`@Service.RunLevel`] actively,
+but via code generated classes -> See <<RunLevel,RunLevel>> for more information.
+This is the only approach that is fully reflection free and skips lookups for injection points.
+- The Helidon startup class `io.helidon.Main`, which will start the registry manager and initialize all
+link:{service-registry-base-url}/io/helidon/service/registry/Service.RunLevel.html[`@Service.RunLevel`]
+services, though it uses service discover (which in turn must use reflection to get service descriptor instances)
+
+=== ServiceRegistryManager
+Manager is responsible for managing the state of a single
+link:{service-registry-base-url}/io/helidon/service/registry/ServiceRegistry.html[`ServiceRegistry`] instance.
+
+When created programmatically, two possible methods can be chosen.
+
+- `create` - Creates a new
+link:{service-registry-base-url}/io/helidon/service/registry/ServiceRegistry.html[`ServiceRegistry`] instance,
+but does not populate it with any services. Service registry is only populated when asked for a specific contract.
+- `start` - Creates a new
+link:{service-registry-base-url}/io/helidon/service/registry/ServiceRegistry.html[`ServiceRegistry`] instance and
+creates all the available services.
+It also properly executes any available
+link:{service-registry-base-url}/io/helidon/service/registry/Service.RunLevel.html[`@Service.RunLevel`] annotations
+present on the service.
+See <<RunLevel,RunLevel>> chapter.
+
+It is important to note, that once you don’t need your service registry,
+method `shutdown` on the manager must be called to ensure proper termination of the service registry.
+
+=== RunLevel
+link:{service-registry-base-url}/io/helidon/service/registry/Service.RunLevel.html[`@Service.RunLevel`]
+is the annotation used for specifying the order in which certain services should be created.
```

Comment:
I would not use "order .. be created". 
The run level provides steps in which an application is started, and within each step (i.e. `@Service.RunLevel(10)`), you may have zero to n services to be started. 
The order of creation is defined by `@Weight`; Run level creates "groups" of services.